File: sdks/go/replication.go
package vectorizer

// GetReplicationStatus returns the current replication status and role of this node.
//
// Calls GET /replication/status.
func (c *Client) GetReplicationStatus() (*ReplicationStatus, error) {
	var result ReplicationStatus
	if err := c.request("GET", "/replication/status", nil, &result); err != nil {
		return nil, err
	}
	return &result, nil
}

// ConfigureReplication configures this node's replication role and parameters.
//
// Calls POST /replication/configure with the provided ReplicationConfig.
// A server restart is required for the new configuration to take effect.
func (c *Client) ConfigureReplication(config *ReplicationConfig) error {
	return c.request("POST", "/replication/configure", config, nil)
}

// GetReplicationStats returns raw replication statistics for the active replication node.
//
// Calls GET /replication/stats. Returns an error when replication is not
// enabled on this node.
func (c *Client) GetReplicationStats() (*ReplicationStats, error) {
	var result ReplicationStats
	if err := c.request("GET", "/replication/stats", nil, &result); err != nil {
		return nil, err
	}
	return &result, nil
}

// ListReplicas returns the replica nodes connected to this master.
//
// Calls GET /replication/replicas. Only available on master nodes; returns
// an error otherwise.
func (c *Client) ListReplicas() ([]ReplicaInfo, error) {
	var result struct {
		Replicas []ReplicaInfo `json:"replicas"`
	}
	if err := c.request("GET", "/replication/replicas", nil, &result); err != nil {
		return nil, err
	}
	return result.Replicas, nil
}

File: sdks/csharp/Replication.cs
using Vectorizer.Models;

namespace Vectorizer;

public partial class VectorizerClient
{
    /// <summary>
    /// Returns the current replication status and role of this node.
    /// Calls GET /replication/status.
    /// </summary>
    public async Task<ReplicationStatus> GetReplicationStatusAsync(
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<ReplicationStatus>(
            "GET", "/replication/status", null, cancellationToken);
    }

    /// <summary>
    /// Configures this node's replication role and parameters.
    /// Calls POST /replication/configure.
    /// A server restart is required for the new configuration to take effect.
    /// </summary>
    public async Task ConfigureReplicationAsync(
        ReplicationConfig config,
        CancellationToken cancellationToken = default)
    {
        await RequestAsync<object>(
            "POST", "/replication/configure", config, cancellationToken);
    }

    /// <summary>
    /// Returns raw replication statistics for the active replication node.
    /// Calls GET /replication/stats.
    /// Returns an error when replication is not enabled on this node.
    /// </summary>
    public async Task<ReplicationStats> GetReplicationStatsAsync(
        CancellationToken cancellationToken = default)
    {
        return await RequestAsync<ReplicationStats>(
            "GET", "/replication/stats", null, cancellationToken);
    }

    /// <summary>
    /// Returns the replica nodes connected to this master.
    /// Calls GET /replication/replicas.
    /// Only available on master nodes; returns an error otherwise.
    /// </summary>
    public async Task<IReadOnlyList<ReplicaInfo>> ListReplicasAsync(
        CancellationToken cancellationToken = default)
    {
        var envelope = await RequestAsync<ReplicasEnvelope>(
            "GET", "/replication/replicas", null, cancellationToken);
        return envelope.Replicas ?? Array.Empty<ReplicaInfo>();
    }

    // Private envelope type for the {replicas: [...]} response shape.
    private sealed class ReplicasEnvelope
    {
        [System.Text.Json.Serialization.JsonPropertyName("replicas")]
        public IReadOnlyList<ReplicaInfo>? Replicas { get; set; }
    }
}
